Src/BootCamp.Chapter/MostCommonLetterFinder.cs

```diff
@@ -1,10 +1,89 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Collections.Generic;
+
+namespace BootCamp.Chapter
 {
     public static class MostCommonLetterFinder
     {
         public static char Find(string sentence)
         {
-            return 'z';
+            if (string.IsNullOrWhiteSpace(sentence))
+            {
+                throw new ArgumentNullException();
+            }
+
+            List<char> sentenceList = new List<char>();
```

Comment:
This list is not needed- a string can be treated as array of char, except that you cannot modify individual character, however you can read individual characters and certainly foreach it.

Reply:
yes i knew that, I did it so I could sort the list. but removed it in the new version with dictionary.